backend/data/src/main/kotlin/io/tolgee/service/InvitationService.kt

```diff
@@ -231,4 +234,15 @@ class InvitationService @Autowired constructor(
   fun userOrInvitationWithEmailExists(email: String, organization: Organization): Boolean {
     return invitationRepository.countByUserOrInvitationWithEmailAndOrganization(email, organization) > 0
   }
+
+  @Transactional
+  fun getInvitationOnRegistration(invitationCode: String?): Invitation? {
+    if (invitationCode == null) {
+      if (!tolgeeProperties.authentication.registrationsAllowed) {
+        throw AuthenticationException(Message.REGISTRATIONS_NOT_ALLOWED)
+      }
```

Comment:
I'm not a fan of this, the code flow is quite obscure here. It's not trivial when observing the sign up function to confirm this check is performed, and I wouldn't consider this function to be responsible for this type of check.

This function receiving a nullable argument to perform this check feels wrong to me as well, you shouldn't encounter cases where "get null" is a valid option and is indicative of a code smell IMHO.

Maybe the naming is just not right? Although I'd consider checking if signing up is allowed something of the concern of the sign up service...